flume-ng-core/src/test/java/org/apache/flume/source/TestStressSource.java

```diff
@@ -82,6 +82,49 @@ public void testMaxTotalEvents() throws InterruptedException,
     }
     verify(mockProcessor, times(35)).processEvent(getEvent(source));
   }
+  
+  @Test
+  public void testRateLimitedEvents() throws InterruptedException,
```

Comment:
This test method covers 2 test cases: rate limited events and non rate limited events.
It would be more straightforward to separate these cases into 2 test methods (1 test case - 1 test method with appropriate names describing the cases).

Reply:
Done. In fact, with the batching changes, now broken into four tests